### ISSUE TITLE
Tenacious Dead returns from exile

### ISSUE THREAD

**Issue by Jipsu** [magefree/mage#11775]
[[Tenacious Dead]] returns to the battlefield even if it's exiled from the graveyard in response to the trigger. Correct behavior would be to allow the last controller to pay the cost of {1}{b} but not return Tenacious Dead if it has changed zones.

Not sure if similar cards work the same: Boneyard Scourge, Lim-Dûl the Necromancer, Reborn Hero, Relentless Dead.

Boneyard Scourge has this additional ruling: 
> If Boneyard Scourge leaves your graveyard before its triggered ability resolves, it won’t be returned to the battlefield. (2017-08-25)

Googling suggests that this delayed trigger rule is the relevant one:
> 603.7c A delayed triggered ability that refers to a particular object still affects it even if the object changes characteristics. However, **if that object is no longer in the zone it’s expected to be in at the time the delayed triggered ability resolves, the ability won’t affect it.** (Note that if that object left that zone and then returned, it’s a new object and thus won’t be affected. See rule 400.7.)

Gamelog:
> 11:59 AM, T10.CD: Tenacious Dead [55b] died
> 11:59 AM, T10.CD: Opponent - Ability triggers: Tenacious Dead [55b] - When Tenacious Dead [55b] dies, you may pay 1B. If you do, return it to the battlefield tapped under its owner's control.
> 12:00 PM, T10.CD: Jipsu activates: Exile target card from a graveyard. If it was a creature card, put a +1/+1 counter on Scavenging Ooze [8a1] and you gain 1 life. from Scavenging Ooze [8a1] targeting Tenacious Dead [55b]
> 12:00 PM, T10.CD: Jipsu moves [55b] from graveyard to the exile zone (source: [8a1])
> 12:00 PM, T10.CD: Jipsu gains 1 life (source: Scavenging Ooze [8a1])
> 12:00 PM, T10.CD: Opponent paid for Tenacious Dead [55b] - Pay 1B and return it to the battlefield tapped under its owner's control?
> 12:00 PM, T10.CD: Opponent puts Tenacious Dead [55b] from exile zone onto the Battlefield

**Comment by JayDi85** [comment]
Yeah, it ignore ZCC data. Whole `ReturnToBattlefieldUnderOwnerControlSourceEffect` must be checked for possible bugged usage and fixed to 0 param instead -1.

![shot_240210_111755](https://github.com/magefree/mage/assets/8344157/3e28fa9b-94c7-4148-90f2-c27f0ba62101)
![shot_240210_111809](https://github.com/magefree/mage/assets/8344157/7f183aa5-4cfb-439b-ad03-887df2b9177f)

Other cards ca be bugged too:
![shot_240210_111954](https://github.com/magefree/mage/assets/8344157/4b6a9938-30cc-44a8-87bc-57cc4dee8176)
